Entities/LobbyMapWarp.cs

```diff
@@ -0,0 +1,104 @@
+using Celeste.Mod.CollabUtils2.UI;
+using Celeste.Mod.Entities;
+using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using Monocle;
+using System.Collections;
+using System.Linq;
+
+namespace Celeste.Mod.CollabUtils2.Entities {
+    [CustomEntity("CollabUtils2/LobbyMapWarp")]
+    public class LobbyMapWarp : Entity {
+        private readonly string spritePath;
+        private readonly bool spriteFlipX;
+        private readonly string activateSpritePath;
+        private readonly bool activateSpriteFlipX;
+        private readonly Facings playerFacing;
+        
+        private LobbyMapController.MarkerInfo info;
+
+        private Sprite sprite;
+
+        public LobbyMapWarp(EntityData data, Vector2 offset) : base(data.Position + offset) {
+            spritePath = data.Attr("spritePath", "decals/1-forsakencity/bench_concrete");
+            spriteFlipX = data.Bool("spriteFlipX");
+            activateSpritePath = data.Attr("activateSpritePath", "CollabUtils2/characters/sitBench");
+            activateSpriteFlipX = data.Bool("activateSpriteFlipX");
+            playerFacing = (Facings) data.Int("playerFacing", (int) Facings.Right);
+            Depth = data.Int("depth", Depths.Below);
+
+            LobbyMapController.MarkerInfo.TryParse(data, null, out info);
+
+            if (!string.IsNullOrWhiteSpace(spritePath)) {
```

Comment:
you could just make the warp sprite be an `Image`, then you could handle flipping with plugin logic (and not have to make it a weird default `Sprite`)